sdk/python/kfp/dsl/_component_bridge.py

```diff
@@ -11,91 +11,187 @@
-        ],
-    )
-    dsl.ContainerOp._DISABLE_REUSABLE_COMPONENT_WARNING = old_warn_value
-
-    component_meta = copy.copy(component_spec)
-    task._set_metadata(component_meta)
-    component_ref_without_spec = copy.copy(component_ref)
-    component_ref_without_spec.spec = None
-    task._component_ref = component_ref_without_spec
-
-    # Previously, ContainerOp had strict requirements for the output names, so we had to
-    # convert all the names before passing them to the ContainerOp constructor.
-    # Outputs with non-pythonic names could not be accessed using their original names.
-    # Now ContainerOp supports any output names, so we're now using the original output names.
-    # However to support legacy pipelines, we're also adding output references with pythonic names.
-    # TODO: Add warning when people use the legacy output names.
-    output_names = [output_spec.name for output_spec in component_spec.outputs or []] # Stabilizing the ordering
-    output_name_to_python = generate_unique_name_conversion_table(output_names, _sanitize_python_function_name)
-    for output_name in output_names:
-        pythonic_output_name = output_name_to_python[output_name]
-        # Note: Some component outputs are currently missing from task.outputs (e.g. MLPipeline UI Metadata)
-        if pythonic_output_name not in task.outputs and output_name in task.outputs:
-            task.outputs[pythonic_output_name] = task.outputs[output_name]
-
-    if container_spec.env:
-        from kubernetes import client as k8s_client
-        for name, value in container_spec.env.items():
-            task.container.add_env_variable(k8s_client.V1EnvVar(name=name, value=value))
-
-    if component_spec.metadata:
-        annotations = component_spec.metadata.annotations or {}
-        for key, value in annotations.items():
-            task.add_pod_annotation(key, value)
-        for key, value in (component_spec.metadata.labels or {}).items():
-            task.add_pod_label(key, value)
-        # Disabling the caching for the volatile components by default
-        if annotations.get('volatile_component', 'false') == 'true':
-            task.execution_options.caching_strategy.max_cache_staleness = 'P0D'
-
-    return task
+) -> ContainerOp:
+
+  pipeline_task_spec = pipeline_spec_pb2.PipelineTaskSpec()
+  pipeline_task_spec.task_info.name = component_spec.name
+
+  # Check types of the reference arguments and serialize PipelineParams
+  arguments = arguments.copy()
+  for input_name, argument_value in arguments.items():
+    if isinstance(argument_value, dsl.ContainerOp):
+      raise TypeError('ContainerOp object was passed to component as an input '
+                      'argument. Pass a single output instead.')
+    elif isinstance(argument_value, dsl.PipelineParam):
+      input_type = component_spec._inputs_dict[input_name].type
+      reference_type = argument_value.param_type
+      dsl.types.verify_type_compatibility(
```

Comment:
Here we're relying on the assumption that `dsl` contains a submodule `types`. A better practice would be directly importing `dsl.types` and only calling `types.verify_type_compatibility` here